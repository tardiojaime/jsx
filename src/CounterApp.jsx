import React, { useState } from "react";
import PropTypes from "prop-types";
export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handle = () => {
        setCounter(counter + 1);
    }
    const handlesub = () =>{
        setCounter((counter)=>counter-1)
    }
    const reset = () =>{
        setCounter(0);
    }
  return (
    <>
      <h1>Counter App</h1>
      <h2>{ counter }</h2>
      <button onClick={handle}>incrementar</button>
      <button onClick={handlesub}>subtract</button>
      <button onClick={reset}>reset</button>
    </>
  );
};
CounterApp.propTypes = {
  value: PropTypes.number,
};
CounterApp.defaultProps = {
  value: 21,
};
