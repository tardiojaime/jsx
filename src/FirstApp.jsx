/* import './style.css'; */
import PropTypes from 'prop-types'
/* 
para no utilizar una etiqueta padre en el componente 
podemos utilizar Fragment de react, una forma mas rapida
es utilizar <></> 
puesto que un componente solo puede retornar un componente
*/
/* 
se recomienda utilizar variable fuera del componente variables 
que no necesitan ser procesados
react no puede imprimir objetos
012 a continuacion
*/
/* 
export const FirstApp = (props) => {
desustructuracion de props  
export const FirstApp = ({title}) => {
*/

export const FirstApp = ({ title, numero }) => {
  return (
    <>
    <h1> { title } </h1>
    <p>Tardio Jaime { numero }</p>
    </>
  )
}
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  numero: PropTypes.number
}
FirstApp.defaultProps = {
  title: 'no hay titulo',
  numero: 25
}
/* 
PropTypes, defenir los props
 */