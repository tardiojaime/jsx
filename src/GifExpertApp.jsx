import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

function GifExpertApp() {

  const [categories, setCategories] = useState(['Dragon Ball']);
  
  const onAddCategory = (newCategory) =>{
    if(categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories])
    //...Desustructuracion
    //setCategories((cat)=>[...cat, 'Pokemon']);
  }
  return (
    <>
    {/* title */}
      <h1>Gif Expert App</h1>
    {/* input */}
    <AddCategory 
    /* setCategories = { setCategories } */
    onNewCategory= { event =>onAddCategory(event)}
    />
    {/* listado de Gif */}
    <ol>
      { categories.map(cate =>(
        <GifGrid key={cate} category={ cate }/>
      ))}
    </ol>
    </>
  )
}

export default GifExpertApp
