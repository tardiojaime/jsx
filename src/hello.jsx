/* 
se recomienda seguir la notacion pascal case
    ** camel case -> contarElementos
    ** pascal case -> ContarElementos
en los nombres de los componentes
 */

export function Hello() {
  return (
    <>
      <h1>hola mundoss</h1>
    </>
  );
}


/* si se exporta de esta forma
la importacion debe ser de la siguiente forma
import componente 'ruta/hello.js'
 ### el componente puede tomar cualquier nombre
 puesto que a de tomar el componente esportado
export default Hello; */

/* 
export function App(){

} 
con este tipo de exportacion la importacion 
es de la siguiente manera
exportacion independiente
import  {Hello}  from './hello'
*/