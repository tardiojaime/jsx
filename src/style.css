h1{
    color: #590a9b;
}