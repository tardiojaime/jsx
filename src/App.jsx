import { useState } from 'react'
import './App.css'
import { FirstApp } from './FirstApp'
function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      <h1>Vite + React</h1>
    <FirstApp/>
    </>
  )
}

export default App
