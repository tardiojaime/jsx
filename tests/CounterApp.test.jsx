import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";
describe("# Pruebas en el Componente <CounterApp />", () => {
    const initValue = 100;
    test("Debe de hacer match con el snapshot", () => {
        const { container } = render(<CounterApp value={initValue} />);
        expect(container).toMatchSnapshot();
    });
    test("Debe de mostrar el valor inical de 100 <CounterApp />", () => {
        render(<CounterApp value={100} />);
        expect(screen.getByText(100)).toBeTruthy();
        //expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100')
    });
    //simulador de eventos...
    test("debe incrementar en uno con el click incrementar", () => {
        const initvalue = 10;
        render(<CounterApp value={initvalue} />);
        //realizamos un evento click
        fireEvent.click(screen.getByText("incrementar"));
        expect(screen.getByText("11")).toBeTruthy();
    });
    test("debe decrementarse en menos 1", () => {
        const counter = 2;
        render(<CounterApp value={counter} />);
        fireEvent.click(screen.getByText("subtract"));
        screen.debug();
        expect(screen.getByText("1")).toBeTruthy();
    });
    test('Debe de reiniciar su valor a cero', () => { 
        const defaultvalue = 90
        render(<CounterApp value={ defaultvalue} />)
        fireEvent.click(screen.getByText('reset'));
        screen.debug()
        expect(screen.getByText('0')).toBeTruthy();
     })
});
