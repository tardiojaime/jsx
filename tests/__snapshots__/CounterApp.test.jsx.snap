// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Pruebas en el Componente <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    incrementar
  </button>
  <button>
    subtract
  </button>
  <button>
    reset
  </button>
</div>
`;
